raiden/network/proxies/token_network.py

```diff
@@ -2213,47 +2239,66 @@ def settle(
+                    gas_limit = safe_gas_limit(gas_limit, GAS_REQUIRED_FOR_SETTLE_CHANNEL)
 
-                transaction_hash = self.proxy.transact(
-                    "settleChannel", gas_limit, channel_identifier=channel_identifier, **kwargs
-                )
-                self.client.poll(transaction_hash)
-                receipt_or_none = check_transaction_threw(self.client, transaction_hash)
+                    transaction_hash = self.proxy.transact(
+                        "settleChannel", gas_limit, channel_identifier=channel_identifier, **kwargs
+                    )
+                    self.client.poll(transaction_hash)
+                    receipt_or_none = check_transaction_threw(self.client, transaction_hash)
 
-        transaction_executed = gas_limit is not None
-        if not transaction_executed or receipt_or_none:
-            if transaction_executed:
-                block = receipt_or_none["blockNumber"]
-            else:
-                block = checking_block
+            transaction_executed = gas_limit is not None
+            if not transaction_executed or receipt_or_none:
+                if transaction_executed:
+                    block = receipt_or_none["blockNumber"]
+                else:
+                    block = checking_block
 
-            self.proxy.jsonrpc_client.check_for_insufficient_eth(
-                transaction_name="settleChannel",
-                address=self.node_address,
-                transaction_executed=transaction_executed,
-                required_gas=GAS_REQUIRED_FOR_SETTLE_CHANNEL,
-                block_identifier=block,
-            )
-            msg = self._check_channel_state_after_settle(
-                participant1=self.node_address,
-                participant2=partner,
-                block_identifier=block,
+                self.proxy.jsonrpc_client.check_for_insufficient_eth(
+                    transaction_name="settleChannel",
+                    address=self.node_address,
+                    transaction_executed=transaction_executed,
+                    required_gas=GAS_REQUIRED_FOR_SETTLE_CHANNEL,
+                    block_identifier=block,
+                )
+                msg = self._check_channel_state_after_settle(
+                    participant1=self.node_address,
+                    participant2=partner,
+                    block_identifier=block,
+                    channel_identifier=channel_identifier,
+                )
+                error_msg = f"{error_prefix}. {msg}"
+                log.critical(error_msg, **log_details)
+                raise RaidenUnrecoverableError(error_msg)
+
+            log.info("settle successful", **log_details)
+        else:
+            log.info("Is light client, storing new state to handle message for settlement.", **log_details)
+            # is LC, we create a new state to store a message for the LC
+            store_settlement_message = ActionStoreSettlementMessage(
```

Comment:
I dont think this action of storing a message is a state change, i think is more an event "store this on the db" (we have the StoreMessageEvent for that)

Also is not a great idea to invoke the handle and track state change from a token network proxy. This class is used to invoke onchain operations with TokenNetwork contract, not to generate new state changes

Reply:
i agree here, it jumped out to me that we now need to pass a `raiden`-typed parameter to this call.

let's please make sure we absolutely cannot do without this